.github/workflows/ci-build.yml

```diff
@@ -10,118 +10,95 @@ on:
-    name: Documentation Style Check
+  doc-style:
+    name: "Documentation style"
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-  
-      - name: Running Vale
-        uses: errata-ai/vale-action@reviewdog
-        env:
-          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+      - name: "Documentation style checks"
+        uses: pyansys/actions/doc-style@v1
         with:
-          files: doc
-          reporter: github-pr-check
-          level: error
-          filter_mode: nofilter
-          fail_on_error: true
-          vale_flags: "--config=doc/.vale.ini"
+          token: ${{ secrets.GITHUB_TOKEN }}
 
-  docs:
-    name: Documentation
-    needs: [style, docs-style]
+  doc-build:
+    name: "Building documentation"
+    needs: [code-style, doc-style]
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - name: "Install mermaid-cli"
+        run: npm install -g @mermaid-js/mermaid-cli
 
-      - name: Set up Python
-        uses: actions/setup-python@v4
+      - name: "Run Ansys documentation building action"
+        uses: pyansys/actions/doc-build@v1
         with:
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
-
-      - name: Install system dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y xvfb texlive-latex-extra latexmk nodejs npm graphviz
-          npm install -g @mermaid-js/mermaid-cli
-
-      - name: Install Python dependencies
-        run: |
-          python -m pip install --upgrade pip tox
-
-      - name: Build HTML documentation
-        run: |
-          xvfb-run tox -e doc
-      - name: Build PDF Documentation
-        run: |
-          sudo apt update
-          sudo apt-get install -y texlive-latex-extra latexmk
-          python -m pip install -r requirements/requirements_doc.txt
-          make -C doc pdf
-
-      - name: Upload HTML documentation
-        uses: actions/upload-artifact@v2.2.4
-        with:
-          name: HTML-Documentation
-          path: .tox/doc_out_html/
-          retention-days: 7
-
-      - name: Upload PDF Documentation
-        uses: actions/upload-artifact@v2.2.4
-        with:
-          name: PDF-Documentation
-          path: doc/build/latex/pyansys*.pdf
-          retention-days: 7
-
-      - name: Deploy to gh-pages
-        if: github.event_name == 'push'
-        uses: JamesIves/github-pages-deploy-action@3.7.1
-        with:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          BRANCH: gh-pages
-          FOLDER: .tox/doc_out_html/
-          CLEAN: true
-          SINGLE_COMMIT: true
-
-  release:
-    if: github.event_name == 'push' && contains(github.ref, 'refs/tags')
-    needs: [style, docs-style, docs]
+          use-python-cache: false
+          skip-install: true
+          requires-xvfb: true
+          dependencies: 'xvfb texlive-latex-extra latexmk nodejs npm graphviz'
+  
+  doc-deploy:
+    name: "Deploy developers documentation"
```

Comment:
@jorgepiloto Can we just make it "Deploy developer documentation"? Technically, we'd want developer's (singular, possessive, but I don't think we need or want apostrophes in here. Although after reading the names of others, maybe we do just put the apostrophe because they are more verbose than I expected. It's up to you.

Reply:
Ops... This shall read "Deploy development documentation".